end_to_end/nlp_mlops_company_sentiment/lambda/lambda_deployer.py

```diff
@@ -0,0 +1,177 @@
+
+    Args:
+        model_package_group_name: The model package group name.
+
+    Returns:
+        The SageMaker Model Package ARN.
+    """
+    try:
+        # Get the latest approved model package
+        response = sagemaker_boto_client.list_model_packages(
+            ModelPackageGroupName=model_package_group_name,
+            ModelApprovalStatus="Approved",
+            SortBy="CreationTime",
+            MaxResults=100,
+        )
+        approved_packages = response["ModelPackageSummaryList"]
+
+        # Fetch more packages if none returned with continuation token
+        while len(approved_packages) == 0 and "NextToken" in response:
+            LOGGER.debug("Getting more packages for token: {}".format(response["NextToken"]))
+            response = sagemaker_boto_client.list_model_packages(
+                ModelPackageGroupName=model_package_group_name,
+                ModelApprovalStatus="Approved",
+                SortBy="CreationTime",
+                MaxResults=100,
+                NextToken=response["NextToken"],
+            )
+            approved_packages.extend(response["ModelPackageSummaryList"])
+
+        # Return error if no packages found
+        if len(approved_packages) == 0:
+            error_message = ("No approved ModelPackage found for ModelPackageGroup: {}".format(model_package_group_name))
+            LOGGER.error("{}".format(error_message))
+
+            raise Exception(error_message)
+
+        model_package = approved_packages[0]
+        LOGGER.info("Identified the latest approved model package: {}".format(model_package))
+
+        return model_package
+    except ClientError as e:
+        stacktrace = traceback.format_exc()
+        error_message = e.response["Error"]["Message"]
+        LOGGER.error("{}".format(stacktrace))
+
+        raise Exception(error_message)
+
+
+def lambda_handler(event, context):
+
+    # Parameters passed in the event 
+    deploy_instance = event["deploy_model_instance_type"]
+    deploy_instance_count = event["deploy_model_instance_count"]
+    endpoint_name = event["endpoint_name"]
+    model_package_group_name = event["model_package_group_name"]
+    role = event["role"]
+    
+    #Get latest approved version of model from model registry
+    model_package_approved = get_approved_package(model_package_group_name)
+    model_package_version = model_package_approved["ModelPackageVersion"]
+    model_package = describe_model_package(model_package_approved["ModelPackageArn"])
+
+    model_name = f'{endpoint_name}-model-v{model_package_version}'
+    ep_config_name = f'{endpoint_name}-epc-v{model_package_version}'
+    
+    # Create a model using the new approved version from registry
+    new_model = sagemaker_boto_client.create_model(ModelName=model_name,
+                                    PrimaryContainer={
+                                                       'Image': model_package["InferenceSpecification"]["Containers"][0]['Image'],
```

Comment:
Style nit - be consistent with either single or double quotes throughout all py files.